src/HDWallet.cpp

```diff
@@ -146,22 +161,36 @@ std::optional<std::string> HDWallet::getAddressFromExtended(const std::string& e
 
 namespace {
     HDNode getNode(const HDWallet& wallet, uint32_t purpose, uint32_t coin) {
-        auto node = getMasterNode(wallet);
+        auto coinIsAion = coin == TWCoinTypeAion;
+        
+        auto node = coinIsAion ? getMasterNodeWithED25519(wallet) : getMasterNode(wallet);
         hdnode_private_ckd(&node, purpose | 0x80000000);
         hdnode_private_ckd(&node, coin | 0x80000000);
         return node;
     }
 
     HDNode getNode(const HDWallet& wallet, uint32_t purpose, uint32_t coin, uint32_t account, uint32_t change, uint32_t address) {
-        auto node = getMasterNode(wallet);
+        auto coinIsAion = coin == TWCoinTypeAion;
+        
+        auto node = coinIsAion ? getMasterNodeWithED25519(wallet) : getMasterNode(wallet);
```

Comment:
Don’t check it by coin type, maybe another getEd2519Node method?

Reply:
Maybe add switch case? there is only a few chains that will use this too: stellar, nano